src/solidity/importresolver.ts

```diff
@@ -32,3 +32,51 @@ export function SolidityImportResolver() {
     .addResolver(UrlResolver())
     .addParser(ImportParser());
 }
+
+export function findImports(data: ImportFile) : string[] {
+  let result: string[] = [];
+  const regex: RegExp = /import\s+(?:(?:"([^;]*)"|'([^;]*)')(?:;|\s+as\s+[^;]*;)|.+from\s+(?:"(.*)"|'(.*)');)/g;
+  let match: RegExpExecArray | null;
+  while (match = regex.exec(data.source)) {
+    for (let i = 1; i < match.length; i++) {
+      if (match[i] !== undefined) {
+        result.push(match[i]);
+        break;
+      }
+    }
+  }
+  return result;
+}
+
+// when importing files from parent directory solc requests sifferent file name
+// e.g. for "../dir/file" solc requests fine named "dir/file"
+// solidifyName returns file name that will  be requested by solc
+export function solidifyName(fileName: string) : string {
+  return fileName.split("./").pop()!;
+}
+
+export async function gatherSources(what: string, workingDir?: string, resolver: ResolverEngine<ImportFile> = SolidityImportResolver()): Promise<ImportFile[]> {
+  let result: ImportFile[] = [];
+  const path = require("path");
+  let queue = [];
+  let alreadyImported = new Set();
+
+  queue.push({cwd: workingDir, file: what});
+  alreadyImported.add(solidifyName(what));
+  while (queue.length > 0){
+    const fileData = queue.shift()!;
+    let tmp: ImportFile = await resolver.require(fileData.file, fileData.cwd);
+    const foundImports = findImports(tmp);
+    result.push(tmp);
+    const filewd = path.dirname(tmp.path);
+    for (let i in foundImports) {
+      const solidifiedName: string = solidifyName(foundImports[i]);
+      if (!alreadyImported.has(solidifiedName)) {
+        alreadyImported.add(solidifiedName);
+        queue.push({cwd: filewd, file: foundImports[i]});
+      }
+    }
+  }
+
+  return result;
```

Comment:
newline - get https://editorconfig.org/ plugin for Visual Studio Code for automatic support of .editorconfig files